Sources/Router/OARoutingHelper.mm

```diff
@@ -617,6 +631,11 @@ - (BOOL) updateCurrentRouteStatus:(CLLocation *)currentLocation posTolerance:(fl
 
 - (CLLocation *) setCurrentLocation:(CLLocation *)currentLocation returnUpdatedLocation:(BOOL)returnUpdatedLocation previousRoute:(OARouteCalculationResult *)previousRoute targetPointsChanged:(BOOL)targetPointsChanged
 {
+    if (![self.class isValidCourseValue:currentLocation.course] && [self.class isValidCourseValue:_app.locationServices.lastKnownHeading])
+    {
+        currentLocation = [[CLLocation alloc] initWithCoordinate:currentLocation.coordinate altitude:currentLocation.altitude horizontalAccuracy:currentLocation.horizontalAccuracy verticalAccuracy:currentLocation.verticalAccuracy course:_app.locationServices.lastKnownHeading speed:currentLocation.speed timestamp:currentLocation.timestamp];
+    }
+    
```

Comment:
Why it needed?

Reply:
Sometimes Location Simulation (and maybe not only it) send there CLLocation with not inited zero course. Which occurs to animations glitches. 

This code helps to detect and ignore that invalid course data, by using last correct value instead.